### ISSUE TITLE
Node cli command not found

### ISSUE THREAD

**Issue by ben-dyer** [nodenv/nodenv#158]
Having trouble running node cli's on pop os 19, nodenv global  14.0.0

For instance in a create-react-app, `npm start` isn't working. I also tried building a simple helloworld-cli.

Steps to reproduce: (1) run `npm init -y `, (2) create `index.js` below, (3) added to package.json: `"bin": { "helloworld": "./index.js" }`, (4) run `npm link`

```js
// index.js
#!/usr/bin/env node
console.log("Hello world!");
```
Running `~/.nodenv/versions/14.0.0/bin/helloworld` works fine. But `helloworld` command is not found.

Is this a nodenv issue or something else? Any help understanding this issue would be much appreciated!

**Comment by ben-dyer** [comment]
Just did a comparison to `nvm` and looks like `nvm` adds
`~/.nvm/versions/node/v14.0.0/bin` to PATH, but nodenv does not(?) put any version specific paths in PATH?

**Comment by jasonkarns** [comment]
@ben-dyer https://github.com/nodenv/nodenv#how-it-works

**Comment by jasonkarns** [comment]
Unsure if related, but something that seems odd to me: why is your `helloworld` bin being placed directly in node's bin/ directory? (ie, `~/.nodenv/versions/14.0.0/bin/helloworld`) For comparison, here's (one of) my node bin directory:

```
$ ls $(nodenv prefix)/bin
node* npm@  npx@
```

**Comment by jasonkarns** [comment]
Ah, sorry, I didn't catch that you were npm-linking it into your global setup.

TL;DR It looks like your npm-linked global package isn't getting a shim created. run `nodenv rehash` once after the npm-link step.


So brief overview from the how-it-works docs is that nodenv inserts shims into your PATH. This is done once so that PATH munging and manipulation does not need to be done as you change-directories around on your system. Thus the "resolution" step of determining the active node version is delayed until a node executable (shim) is actually invoked. This is a safety and performance improvement over "real-time PATH munging"

The tradeoff is that shims need to be created for every node executable. Out of the box, these are `node`, `npm`, and `npx`. Additionally, a shim would need created for each bin provided by globally-installed npm packages. If you have nodenv configured to run/eval nodenv-init, then nodenv-rehash is run once per shell startup.

So: run `nodenv rehash` once after installing (or linking) global packages; or alternatively, spawning a new shell would have done that for you automatically

There is also a nodenv plugin that runs nodenv-rehash after an npm package is installed globally:
https://github.com/nodenv/nodenv-package-rehash (on modern nodes, this plugin is too noisy to ship with nodenv out of the box)
